File: flutter_app/lib/screens/random_quote_screen.dart
import 'dart:math';

import 'package:flutter/material.dart';

import 'package:video_player/video_player.dart';

import 'package:flutter_app/models/quote.dart';
import 'package:flutter_app/services/quotes_service.dart';
import 'package:flutter_app/widgets/quote_card.dart';

class RandomQuoteScreen extends StatefulWidget {
  const RandomQuoteScreen({super.key});

  @override
  State<RandomQuoteScreen> createState() => _RandomQuoteScreenState();
}

class _RandomQuoteScreenState extends State<RandomQuoteScreen> {
  final QuotesService _quotesService = QuotesService();
  late VideoPlayerController _videoPlayerController;
  Quote? _currentQuote;
  bool _isLoading = true;

  @override
  void initState() {
    super.initState();
    _loadRandomQuote();
  }

  Future<void> _loadRandomQuote() async {
    setState(() {
      _isLoading = true;
    });
    _videoPlayerController =
        VideoPlayerController.asset('assets/videos/cinema.mp4')
          ..initialize().then((_) {
            setState(() {
              _videoPlayerController.play();
              _videoPlayerController.setVolume(0);
            });
          });

    try {
      final quotes = await _quotesService.getQuotes();
      if (quotes.isNotEmpty) {
        final random = Random();
        final randomIndex = random.nextInt(quotes.length);
        setState(() {
          _currentQuote = quotes[randomIndex];
        });
      }
    } catch (e) {
      // Handle error
      print('Error loading random quote: $e');
    } finally {
      setState(() {
        _isLoading = false;
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Padding(
          padding: const EdgeInsets.all(16.0),
          child: Column(
            mainAxisAlignment: MainAxisAlignment.start,
            children: [
              _videoPlayerController.value.isInitialized
                  ? Semantics(
                      label: "Kino-Video mit Szenen aus verschiedenen Filmen",
                      child: SizedBox(
                        height: 300,
                        child: VideoPlayer(_videoPlayerController),
                      ),
                    )
                  : const SizedBox.shrink(),
              // Random Quote Card
              if (_isLoading)
                const Center(child: CircularProgressIndicator())
              else if (_currentQuote != null)
                QuoteCard(quote: _currentQuote!)
              else
                Card(
                  margin: const EdgeInsets.symmetric(
                    horizontal: 16,
                    vertical: 8,
                  ),
                  child: Padding(
                    padding: const EdgeInsets.all(24),
                    child: Column(
                      children: [
                        Icon(
                          Icons.format_quote,
                          size: 50,
                          color: Theme.of(context).colorScheme.onSurfaceVariant,
                        ),
                        const SizedBox(height: 16),
                        Text(
                          'Keine Zitate vorhanden',
                          style: TextStyle(
                            fontSize: 18,
                            fontWeight: FontWeight.bold,
                            color: Theme.of(context).colorScheme.onSurface,
                          ),
                        ),
                        const SizedBox(height: 8),
                        Text(
                          'Fügen Sie Ihr erstes Zitat hinzu!',
                          style: TextStyle(
                            color: Theme.of(context).colorScheme.onSurface,
                          ),
                        ),
                      ],
                    ),
                  ),
                ),

              const SizedBox(height: 24),

              // Refresh Button (Get Another Random Quote)
              if (_currentQuote != null)
                SizedBox(
                  width: double.infinity,
                  child: ElevatedButton.icon(
                    onPressed: _loadRandomQuote,
                    icon: const Icon(Icons.refresh),
                    label: const Text('Anderes Zitat', style: TextStyle()),
                    style: OutlinedButton.styleFrom(
                      padding: const EdgeInsets.symmetric(vertical: 16),
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(8),
                      ),
                    ),
                  ),
                ),
            ],
          ),
        ),
      ),
    );
  }
}

File: kmp_app/composeApp/src/commonMain/kotlin/org/example/project/screens/RandomQuoteScreen.kt
package org.example.project.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.FormatQuote
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinx.coroutines.launch
import org.example.project.models.Quote
import org.example.project.services.QuotesService
import org.example.project.widgets.QuoteCard
import kotlin.random.Random

@Composable
fun RandomQuoteScreen(
    quotesService: QuotesService = QuotesService()
) {
    var currentQuote by remember { mutableStateOf<Quote?>(null) }
    var isLoading by remember { mutableStateOf(true) }
    var quotes by remember { mutableStateOf<List<Quote>>(emptyList()) }

    val scope = rememberCoroutineScope()

    suspend fun loadRandomQuote() {
        try {
            isLoading = true
            quotes = quotesService.getQuotes()
            if (quotes.isNotEmpty()) {
                val randomIndex = Random.nextInt(quotes.size)
                currentQuote = quotes[randomIndex]
            } else {
                currentQuote = null
            }
        } catch (e: Exception) {
            // Handle error
            currentQuote = null
        } finally {
            isLoading = false
        }
    }

    LaunchedEffect(Unit) {
        loadRandomQuote()
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(24.dp)
    ) {
        // Video placeholder (since video player is platform-specific)
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .height(300.dp),
            shape = RoundedCornerShape(12.dp),
            colors = CardDefaults.cardColors(containerColor = Color.Black)
        ) {
            Box(
                modifier = Modifier.fillMaxSize(),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = "Video Background",
                    color = Color.White,
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Medium
                )
            }
        }

        // Random Quote Card
        when {
            isLoading -> {
                Box(
                    modifier = Modifier.fillMaxWidth(),
                    contentAlignment = Alignment.Center
                ) {
                    CircularProgressIndicator()
                }
            }
            currentQuote != null -> {
                QuoteCard(
                    quote = currentQuote!!,
                    onTap = null // Disable tap for random screen
                )
            }
            else -> {
                Card(
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(12.dp)
                ) {
                    Column(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(24.dp),
                        horizontalAlignment = Alignment.CenterHorizontally
                    ) {
                        Icon(
                            imageVector = Icons.Default.FormatQuote,
                            contentDescription = null,
                            modifier = Modifier.size(50.dp),
                            tint = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f)
                        )
                        Spacer(modifier = Modifier.height(16.dp))
                        Text(
                            text = "Keine Zitate vorhanden",
                            fontSize = 18.sp,
                            fontWeight = FontWeight.Bold,
                            color = MaterialTheme.colorScheme.onSurface
                        )
                        Spacer(modifier = Modifier.height(8.dp))
                        Text(
                            text = "Fügen Sie Ihr erstes Zitat hinzu!",
                            color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
                        )
                    }
                }
            }
        }

        // Refresh Button (Get Another Random Quote)
        if (currentQuote != null) {
            Button(
                onClick = {
                    scope.launch {
                        loadRandomQuote()
                    }
                },
                modifier = Modifier.fillMaxWidth(),
                enabled = !isLoading
            ) {
                Icon(Icons.Default.Refresh, contentDescription = null)
                Spacer(modifier = Modifier.width(8.dp))
                Text("Anderes Zitat")
            }
        }
    }
}